### PR TITLE
feat/speed-performance

### PR DESCRIPTION
**Pull request by RecurvedBow**

### CODE REVIEW

**Comment by RecurvedBow** [comment]
Closes #125

File: tests/tracing/test_trace_all.py [hunk 1]
```diff
@@ -18,8 +18,8 @@ def main():
 
 
 def test_everything_is_traced():
-    called = decorators.entrypoint(None)(main)
-    assert called is not None
+    trace_data, _ = decorators.entrypoint(None)(main)
```
Comment (bengsparks) [inline tests/tracing/test_trace_all.py:21]: What does this do? the `entrypoint` still returns only a `DataFrame`, right?

**Review by bengsparks — CHANGES_REQUESTED** [review]
Does this branch still contain the changed interfaces I mentioned?

File: tests/tracing/test_trace_all.py (hunk 1 above)
Comment (RecurvedBow) [reply tests/tracing/test_trace_all.py:21]: Additionally, we also return the performance data. This is required for test_evaluation_decorators.py.

**Comment by bengsparks** [comment]
The result of our discussion: 
Move the benchmark attribute into the config file:

```toml
[pytypes]
name = ...
project_path = ...
stdlib_path = ...
venv_path = ...

benchmark = true
```
should make the annotations perform benchmarking. Setting `benchmark` to `false`, or omitting it should make "standard" behaviour occur, i.e. writing traced data to storage

**Review by bengsparks — CHANGES_REQUESTED** [review]
Final feedback:

* `TracerBase` should derive from `abc.ABC`, and should mark its tracing method with `abc.abstractmethod`.
* The current `TracerBase` implementation should become a new class called `NoOpTracer` or `NoOperationTracer`, which derives from the newly abstract `TracerBase` class, and assumes its implementation of the tracing method.
* All tests should use the new tracer instead of the `TracerBase` class.

**Review by bengsparks — APPROVED** [review]

**Pull request merged**